solver/src/interactions/weth.rs

```diff
@@ -1,13 +1,33 @@
 use crate::{encoding::EncodedInteraction, settlement::Interaction};
+use anyhow::{anyhow, Context as _, Result};
 use contracts::WETH9;
 use primitive_types::U256;
 
-#[derive(Debug)]
+#[derive(Clone, Debug)]
 pub struct UnwrapWethInteraction {
     pub weth: WETH9,
     pub amount: U256,
 }
 
+impl UnwrapWethInteraction {
+    /// Tries to merge the specified unwrap with the current one, returning
+    /// `true` if the merge was successful, and `false` otherwise.
+    ///
+    /// Returns an error on arithmetic overflow.
```

Comment:
nanonit: I understand arithmetic overflows are practically impossible. However instead of handling the error case, we could just return false to encode it as its own transaction?

Reply:
> I understand arithmetic overflows are practically impossible

I think they should be impossible on the networks we support given the native currency ERC20 token implementations (WETH and wxDAI). They could happen on test-networks or unsupported configurations.

I just thought that since `SettlementHandling` already returns a `Result` that I might as well propagate the arithmetic overflow. Alternatively we can `panic` as a way to say "we really don't expect this to happen ever" so that we get a louder alert so we can 

Personally, I would prefer panicking over treating it in the same way we do if the `merge` was not possible (i.e. the token address was different).

WDYT?

Reply:
Panicing sounds ok to me in that case (I'm fine either way though)